app/models/host_report.rb

```diff
@@ -56,6 +57,19 @@ def self.search_by_keyword(_key, operator, value)
     }
   end
 
+  def self.summarise(_time = 1.day.ago, *hosts, format)
+    list = {}
+    raise ::Foreman::Exception, N_("invalid host list") unless hosts
+    hosts.flatten.each do |host|
+      host_report_id = HostReport.where(host_id: host.id, format: HostReport.formats[format]).maximum(:id)
+      next if host_report_id.nil?
+      report = HostReport.find(host_report_id)
+      data = { change: report.change, nochange: report.nochange, failure: report.failure }
+      list[host.name] = { :metrics => data, :id => host.id }
+    end
```

Comment:
Variable `hosts` is an ActiveRecord::Relation object, you convert it to array allocating a lot of unnecessary memory. You should just iterate the query result here, preferably with the most efficient `find_each`:

```
hosts.all.find_each do |host|
end
```

Please calculate the total number in here instead of the method you call this method from.